setup.py

```diff
@@ -0,0 +1,44 @@
+from setuptools import setup, find_packages, find_namespace_packages
+import pathlib
+
+here = pathlib.Path(__file__).parent.resolve()
+
+# workaround a bug introduced by pyproject.toml
+# https://github.com/pypa/pip/issues/7953#issuecomment-645133255
+import site, sys; site.ENABLE_USER_SITE = True
+
+setup(
+  name='spinalcordtoolbox-data-<dataset>',
+  description='Part of https://github.com/neuropoly/spinalcordtoolbox',
+  long_description=(here / 'README.md').read_text(encoding='utf-8'),
+  long_description_content_type='text/markdown',
+  author='Neuropoly',
+  author_email='neuropoly@googlegroups.com',
+  url='https://spinalcordtoolbox.com/',
+  project_urls={
+      'Source': 'https://github.com/sct-data/<dataset>',
+      #'Documentation': '',
+  },
+  #license='CC-BY-NC', ??
+  #license_files=[ ... ] # TODO?
+
+  packages=find_namespace_packages('src/'),
+  package_dir={"":"src/"},
+
+  # with setuptools_scm, means it includes non-python files if they're under git
+  include_package_data=True,
+
+  # with setuptools_scm, get the version out of the most recent git tag.
+  # the tags must be formatted as semver.
+  use_scm_version=True,
```

Comment:
This can also be mirrored in `pyproject.toml`, right? From https://pypi.org/project/setuptools-scm/: 

> To enable version inference, add this section to your pyproject.toml:
> 
> ```toml
> # pyproject.toml
> [tool.setuptools_scm]
> ```
> 
> Including this section is comparable to supplying use_scm_version=True in setup.py. Additionally, include arbitrary keyword arguments in that section to be supplied to get_version(). For example:
> 
> ```toml
> # pyproject.toml
> [tool.setuptools_scm]
> write_to = "pkg/version.py"
> ```

Reply:
Yes... but I am rebelling against pyproject.toml along with 70% of python.

I gave a shot to putting everything in pyproject.toml but it doesn't work! It doesn't support editable installs, and doesn't support ...uh I think `packages=find_namespace_packages('src/')`?

So I gave up and moved everything back to setup.py. I would rather keep everything uniquely in setup.py until the day pyproject.toml can fully replace it in one step. I wouldn't even have it except that, at least with the latest pip, with it `pip install .` produces a .dist-info directory, but `pip install -e .` produces a .egg-info directory.

Maybe I should put a comment about this inside pyproject.toml.

Reply:
In that case, since the only content inside `pyproject.toml` is already duplicated in `setup.py`, couldn't we avoid having a `pyproject.toml` at all?